File: src/tut_lab1/src/publisher.py
#!/usr/bin/env python 
import rospy 
from std_msgs.msg import Float32
import random as rd
class RandomAgePublisher:
	def __init__(self):
		self.pub = rospy.Publisher("/randomAge", Float32, queue_size = 1)
		self.age_range = 100
		print("initialized")

		self.publishAge()
	def publishAge(self):
		while not rospy.is_shutdown():
			age = float(rd.choice(range(self.age_range)))
			print(age)
			self.pub.publish(age)


def main():
	rospy.init_node("RandomAgePublisher")
	RandomAgePublisher()
	rospy.spin()

if __name__ == '__main__':
	rospy.init_node("RandomAgePublisher")
	node = RandomAgePublisher()
	rospy.spin()

File: src/cpp_tutorial/src/publisher.cpp
#include <sstream>
#include <iostream>
#include "ros/ros.h"
#include "std_msgs/String.h"
using namespace std;
class Simple_pub{
	private:
		int range;
		
		
		ros::NodeHandle node;
		
	public:
		ros::Publisher pub = node.advertise<std_msgs::String> ("/randomAge", 100);
		Simple_pub(int);
		Simple_pub();

		int age(void){
			return (range);
		}


};
Simple_pub::Simple_pub(int rng){
	range = rng;
}

Simple_pub::Simple_pub(){
	range = 100;
}


int main(int argc, char **argv){
	ros::init(argc, argv, "Simple_pub");
	Simple_pub spub (120);

	while (ros::ok()){
		std_msgs::String msg;
		stringstream str;
		str<< "string to publish"<< endl;
		msg.data = str.str();
		spub.pub.publish(msg);
		ros::spinOnce();
	}
	
	return 10;

}